src/System.Runtime/tests/System/ArrayTests.cs

```diff
@@ -533,6 +550,29 @@ public static IEnumerable<object[]> Copy_TestData()
         }
 
         [Theory]
+        public static void ForEach()
+        {
+            Array.ForEach<int>(new int[] { }, new Action<int>(i => { throw new InvalidOperationException(); }));
+            // Did not throw; no items
+
+            int counter = 0;
+            Array.ForEach<int>(new int[] { 1, 2, 3 }, new Action<int>(i => counter += i));
+            Assert.Equal(6, counter);
+
+            // Only works on one dimensional arrays
+        }
+
+        [Fact]
+        public static void ForEach_Invalid()
+        {
+            Assert.Throws<ArgumentNullException>(() => { Array.ForEach<short>(null, new Action<short>(i => i++)); });  // Array is null
+            Assert.Throws<ArgumentNullException>(() => { Array.ForEach<string>(new string[] { }, null); }); // Action is null
```

Comment:
When used with argument exceptions, Assert.Throws has overloads that take a string parameter name as the first argument, and the assert will compare that against the parameter name from the thrown exception object.  It'd be good to specify that in these calls.